packages/deployer/src/compiler.ts

```diff
@@ -109,9 +126,14 @@ export class Compiler {
             shouldCompile = true;
         } else {
             const currentArtifact = currentArtifactIfExists as ContractArtifact;
-            shouldCompile =
-                currentArtifact.networks[this._networkId].optimizer_enabled !== this._optimizerEnabled ||
-                currentArtifact.networks[this._networkId].source_tree_hash !== sourceTreeHashHex;
+            if (_.isUndefined(currentArtifact.versions) || _.isUndefined(currentArtifact.versions[this._versionName])) {
+                shouldCompile = true;
+            } else {
+                shouldCompile =
```

Comment:
Let's assign the individual conditionals to clear variable names.

E.g
```
const isLegacyArtifact = currentArtifact.schemaVersion !== '2.0.0';
```

Looking at that, shouldn't we do a semantic versioning comparison? What about `2.0.1`, and higher?